mixture_of_experts_pretraining/README.md

```diff
@@ -0,0 +1,515 @@
+
+To obtain the logs, establish an SSH connection to a virtual machine and retrieve the Docker container logs.
+```
+gcloud alpha compute tpus tpu-vm ssh ${TPU_NAME} --project ${PROJECT_ID} --zone ${ZONE} --worker=0
+```
+
+Run `docker logs` to retrieve back logs.
+```
+sudo docker logs -f <container>
+```
+
+## Dry-run Tests
+To perform dry-run tests with different models/parameters on a smaller scale like `v4-8`, use the following python commands as workload:
+### Test with a smaller mixtral model
+```
+python run_clm.py model.config_path=mixtral80.json eval_frequency=3 n_eval_examples=100 per_device_train_batch_size=4 max_steps=30
+``` 
+### Test with a gpt2 model
+```
+python run_clm.py model.name_or_path=gpt2 eval_frequency=3 n_eval_examples=100 per_device_train_batch_size=4 gradient_accumulation_steps=2 sched.warmup_ratio=0. max_steps=30
+```
+
+# 6. Training Mixtral 8x22B with NeMo on GPU Device
+
+**IMPORTANT** GPU implementation is a supplementary reference and it is not used for RCP
+generation. There is convergence gap between GPU and TPU reference and GPU code cannot be used as a
+inplace substitute of TPU code.
+
+## Docker Image
+
+Build and push docker image:
+
+```shell
+docker build -t <registry_path_image_name>:<image_tag> -f docker/gpu/Dockerfile .
+docker push <registry_path_image_name>:<image_tag>
+```
+
+## Kubernetes workflow
+### Run workflow
+
+In order for this workflow to function, in the ```helm-context``` directory, there must exist a **_select-configuration.yaml_** file.
+
+Package and schedule job. An example job name could be "nemo-gpt3-175b-nemo-16gpus". Use whatever is convenient when searching for later.
+
+
+```shell
+helm install <username_workload_job_name> helm-context/
+```
+
+### Monitor workflow
+
+Check pod status (use this to find the name of the pod you want logs from)
+
+
+```shell
+kubectl get pods | grep "<some_part_of_username_workload_job_name>"
+```
+
+
+Check job status
+
+
+```shell
+kubectl get jobs | grep "<some_part_of_username_workload_job_name>"
+```
+
+
+Get logs (Using pod name from earlier)
+
+
+```shell
+kubectl logs "<pod_name>"
+```
+
+## Slurm/Pyxis workflow
+
+### Preprocessing
+
+For GPU implementation both dataset and checkpoint has to be preprocessed. This can be done once,
+before experimentation and saved. **IMPORTANT** saved checkpoint and dataset has to be accessible by
+all nodes in the system.
+
+To get preprocessed checkpoint, run checkpoint_download.py script
+```shell
+python scripts/gpu/checkpoint_download.py --checkpoint_id mistralal/Mixtral-8x22B-v0.1 \
+    --output_dir <path to save checkpoint> --hf_token <your token to HF repository>
+```
+
+This script will download specified checkpoint from huggingface repository, preprocess it and save
+into specified directory
+
+To preprocess dataset, use dataset_preprocessing.py script
```

Comment:
Same as checkpoint

Reply:
Preprocessed dataset can be downloaded from mlcommons bucket. I added annotation while keeping manual preprocessing step in the documentation